### DISCUSSION TITLE
subscribing to dynamic topics

### DISCUSSION THREAD

**Discussion by artknight** [uNetworking/uWebSockets#1260] [Q&A]
I am trying to allow the users to subscribe to dynamic topics like below

```
const app = uWS.App().ws('/ws/:area/:channel', {
   open: ws => {
      //not working
      let area = ws.getParameter(0);

      ws.subscribe(`${area}-${channel}`);
   }
}
```

In the docs ( https://github.com/uNetworking/uWebSockets/blob/master/misc/READMORE.md ) it says specifically ...and i quote `Every websocket route has the same pattern and pattern matching as for Http` which should allow me to get those params.

Note: I am running v19.2.0

Can anyone have any ideas?

**Comment by hst-m** [comment]
you are in the wrong repo JS is other repo https://github.com/uNetworking/uWebSockets.js there is no ws.getParameter it is req.getParameter, you need to use the upgrade handler to pass value from req https://github.com/uNetworking/uWebSockets.js/discussions/339
